src/ucp/core/ucp_worker.c

```diff
@@ -2835,6 +2839,61 @@ static ucs_status_t ucp_worker_address_pack(ucp_worker_h worker,
                             address_length_p, (void**)address_p);
 }
 
+
+static size_t ucp_worker_get_max_am_header(ucp_worker_h worker)
+{
+    ucp_context_h context = worker->context;
+    uct_iface_attr_t *if_attr;
+    ucp_rsc_index_t iface_id;
+    size_t max_am_header, max_uct_fragment;
+    size_t max_rts_size, max_ucp_header;
+
+    if (worker->max_am_header != UCP_WORKER_MAX_AM_HEADER_UNINIT) {
```

Comment:
maybe just calculate it during ucp worker initialization?

Reply:
done